02-peer-protocol.md

```diff
@@ -50,10 +67,361 @@ pubkey corresponding to the funding output nothing prevents duplicative channel
 ids.
 
 
+### `channel_id`, v2
+
+For channels established using the v2 protocol, the `channel_id` is the
+SHA256(lesser-revocation-basepoint || greater-revocation-basepoint),
+where the lesser and greater is based off the order of the basepoint.
+The basepoints are compact DER-encoded public keys.
+
+If the peer's revocation basepoint is unknown (e.g. `open_channel2`),
+a temporary `channel_id` should be found by using a zeroed out basepoint for the unknown peer.
+
+#### Rationale
+These values must be remembered by both peers for correct operation anyway.
+They're known from the first exchange of messages, obviating the need for
+a `temporary_channel_id`. Finally, by mixing information from both sides,
+they avoid `channel_id` collisions.
```

Comment:
Reusing a revocation basepoint would make your revocation points known ahead of time. This would be a huge security hole,as it would make any commitment tx that used a re-used basepoint unsafe. The prior peer would know the key to the revocation path of any published commitment tx (provided their counter went as high as the published one etc).

Reply:
Which is to say, if your counterparty is reusing revocation basepoints for channels, they've got other problems.